src/screen/ProductPage.tsx

```diff
@@ -1,61 +1,106 @@
+};
+
 export function ProductPage(props: ProductPageProps) {
-  const [inputProduct, setInputProduct] = useState<Product>({} as Product);
+  const [inputProduct, setInputProduct] = useState<ProductForm>({
+    title: "",
+    categoryId: 1,
+    price: 0,
+  });
+  // const [product, setProduct] = useState<Product | null>(null); // contoh
   const [inputType, setInputType] = useState<InputType>("add");
-  const [inputCategoryProduct, setInputCategoryProduct] = useState<number>();
-  const { products, reFetchProduct, isLoadingFetchProduct } = useFetchProduct();
-  const [editProduct, setEditProduct] = useState<number>(
-    products ? products[products.length - 1].id + 1 : 1
-  );
-  const { categories, reFetch, isLoadingFetch } = useFetchCategories();
-  const { isLoadingCreate, submit } = useCreateProduct();
-  const { delProduct, isLoadingDelete } = useDeleteProduct();
-  const { isLoadingEdit, updateProduct } = useEditProduct();
+  const [idEditProduct, setIdEditProduct] = useState<number | null>(null);
+  const fetchProduct: FetchProduct = useFetchProduct();
+  const fetchCategories: FetchCategories = useFetchCategories();
+  const createProduct: CreateProduct = useCreateProduct();
+  const deleteProduct: DeleteProduct = useDeleteProduct();
+  const editProduct: EditProduct = useEditProduct();
 
-  function onAddProduct(props: Product) {
-    const id = products ? products[products.length - 1].id + 1 : 1;
-    const newProduct = { ...props, id: id };
-    submit(newProduct).then(() => {
-      reFetchProduct();
+  function onAddProduct(props: ProductForm) {
+    const newProduct = { ...props };
+    createProduct.submit(newProduct).then(() => {
+      setInputProduct({
+        title: "",
+        categoryId: 1,
+        price: 0,
+      });
+      fetchProduct.reFetch();
     });
   }
 
-  function onEditProduct(props: Product) {
-    updateProduct({
-      id: props.id,
-      title: props.title,
-      price: props.price,
-      categoryId: props.categoryId,
-    }).then(() => {
-      const blankProduct = {} as Product;
-      setInputProduct(blankProduct);
-      reFetchProduct();
-    });
+  function onEditProduct(props: ProductForm) {
+    // if (idEditProduct) {
+    if (!idEditProduct) return;
+    editProduct
+      .updateProduct({
+        id: idEditProduct,
+        title: props.title,
+        price: props.price,
+        categoryId: props.categoryId,
+      })
+      .then(() => {
+        setInputType("add");
+        setInputProduct({
+          title: "",
+          categoryId: 1,
+          price: 0,
+        });
+        setIdEditProduct(null);
+        fetchProduct.reFetch();
+      });
+    // }
   }
 
-  function categoryTitle(props: Product) {
-    const nameCategory = categories?.find(
-      (category) => category.id === props.categoryId
+  function categoryTitle(categoryId: number) {
+    const nameCategory = fetchCategories.categories.find(
+      (category: Category) => category.id === categoryId
```

Comment:
ini bisa disingkat jadi kayak gini : 

```suggestion
    return fetchCategories.categories.find((category) => category.id === categoryId)?.title
```

Reply:
updated